### PR TITLE
[#13] [Backend] As a user, I can see the user's info on the Dashboard page

### LINKED ISSUE #13: [Backend] As a user, I can see the user's info on the Dashboard page
## Why
When the user sign in, he should see the Dashboard page with the user's info.

## Acceptance Criteria
- Add GET `/api/v1/me` API
- Write unittest correspondingly.

### PR DESCRIPTION
**Pull request by manh-t**
Close #13

## What happened 👀

- Implement GET `/me` to get the user info
- Prepare `User` data for integrate task

## Insight 📝

N/A

## Proof Of Work 📹

image

### CODE REVIEW

**Comment by manh-t** [comment]
🚀 Deployed on https://64ec212750048d50d8468875--manh-react-survey.netlify.app

**Comment by manh-t** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Congrats, coverage is good</td>
    </tr>
  </tbody>
</table>

## Coverage in All Files
File | Line Coverage | Statement Coverage | Function Coverage | Branch Coverage
---- | ------------: | -----------------: | ----------------: | --------------:
[src/../Dashboard/index.tsx](../blob/0358d76/src/screens/Dashboard/index.tsx) | (1/6) 17% | (1/6) 17% | (0/4) 0% | (0/0) NaN%
[src/../v1/requestManager.ts](../blob/0358d76/src/lib/request/v1/requestManager.ts) | (9/18) 50% | (9/18) 50% | (4/6) 67% | (1/3) 33%
[src/config/index.ts](../blob/0358d76/src/config/index.ts) | (5/9) 56% | (5/9) 56% | (0/3) 0% | (0/4) 0%
[src/tests/error.ts](../blob/0358d76/src/tests/error.ts) | (2/3) 67% | (2/3) 67% | (1/2) 50% | (1/3) 33%
[src/screens/SignIn/index.tsx](../blob/0358d76/src/screens/SignIn/index.tsx) | (12/16) 75% | (12/16) 75% | (3/6) 50% | (8/8) 100%
[src/../Content/index.tsx](../blob/0358d76/src/components/Dashboard/Content/index.tsx) | (8/10) 80% | (8/10) 80% | (4/6) 67% | (3/5) 60%
[src/../index.tsx](../blob/0358d76/src/routes/ProtectedRoute/index.tsx) | (4/5) 80% | (4/5) 80% | (2/2) 100% | (1/2) 50%
[src/App.tsx](../blob/0358d76/src/App.tsx) | (3/3) 100% | (3/3) 100% | (1/1) 100% | (0/0) NaN%
[src/index.tsx](../blob/0358d76/src/index.tsx) | (3/3) 100% | (3/3) 100% | (0/0) NaN% | (0/0) NaN%
[src/../Authentication/index.ts](../blob/0358d76/src/adapters/Authentication/index.ts) | (2/2) 100% | (2/2) 100% | (1/1) 100% | (0/0) NaN%
[src/adapters/Base/index.ts](../blob/0358d76/src/adapters/Base/index.ts) | (8/8) 100% | (8/8) 100% | (2/2) 100% | (2/2) 100%
[src/adapters/BaseAuth/index.ts](../blob/0358d76/src/adapters/BaseAuth/index.ts) | (10/10) 100% | (10/10) 100% | (4/4) 100% | (2/2) 100%
[src/adapters/Survey/index.ts](../blob/0358d76/src/adapters/Survey/index.ts) | (1/1) 100% | (1/1) 100% | (1/1) 100% | (0/0) NaN%
[src/adapters/User/index.ts](../blob/0358d76/src/adapters/User/index.ts) | (1/1) 100% | (1/1) 100% | (1/1) 100% | (0/0) NaN%
[src/components/Alert/index.tsx](../blob/0358d76/src/components/Alert/index.tsx) | (3/3) 100% | (3/3) 100% | (2/2) 100% | (0/0) NaN%
Other (24 more) | (101/101) 100% | (101/101) 100% | (35/35) 100% | (30/31) 97%
Total | (173/199) 87% | (173/199) 87% | (61/76) 80% | (48/60) 80%

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0bce7dc60c61d4ca8af568ab14cde75b36f826ab
</p>

File: src/adapters/User/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { getAuth } from 'adapters/BaseAuth';
+
+import { getUserInfo } from '.';
+
+jest.mock('adapters/BaseAuth');
+
+describe('UserAdapter', () => {
+  beforeEach(() => {
+    (getAuth as jest.Mock).mockImplementation(() => jest.fn());
```
Comment (nvminhtue) [inline src/adapters/User/index.test.ts]: IMO I don't think we need to mock this function

File: src/adapters/User/index.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { getAuth } from 'adapters/BaseAuth';
+
+export const getUserInfo = () => getAuth('me');
```
Comment (nvminhtue) [inline src/adapters/User/index.ts:3]: is it supposed to be?
```suggestion
export const getUserInfo = async (): Promise<User> => getAuth('me');
```

Comment (manh-t) [reply src/adapters/User/index.ts:3]: @nvminhtue the return type is `Promise<AxiosResponse<any, any>>`, we haven't deserialized it to `User` yet. So I think we can write it shorter like above, or:
```suggestion
export const getUserInfo = (): Promise<AxiosResponse<any, any>> => getAuth('me');
```

File: src/adapters/User/index.test.ts (hunk 1 above)
Comment (manh-t) [reply src/adapters/User/index.test.ts]: @nvminhtue thanks, I removed it in [dc2b9c2](https://github.com/manh-t/react-survey/pull/59/commits/dc2b9c2910ddc2065bd102c378760f03848fdadd)

File: src/adapters/User/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { getAuth } from 'adapters/BaseAuth';
+
+import { getUserInfo } from '.';
+
+jest.mock('adapters/BaseAuth');
+
+describe('UserAdapter', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
```
Comment (tyrro) [inline src/adapters/User/index.test.ts]: do we need this `afterEach` block for this test?

**Review by tyrro — APPROVED** [review]

File: src/adapters/User/index.test.ts (hunk 3 above)
Comment (manh-t) [reply src/adapters/User/index.test.ts]: @tyrro not sure if I understand correctly, but each time we do mock `jest.mock('adapters/BaseAuth');` at line 5, we should clear mock each test, is that correct? 😅

Comment (hoangmirs) [reply src/adapters/User/index.test.ts]: I think we don't need it. There are some tips for using `clear/reset/restore` mock:
https://stackoverflow.com/a/66361187

**Review by hoangmirs — APPROVED** [review]

File: src/adapters/User/index.test.ts (hunk 3 above)
Comment (manh-t) [reply src/adapters/User/index.test.ts]: Thanks a lot, @hoangmirs. I removed it in [0bce7dc](https://github.com/manh-t/react-survey/pull/59/commits/0bce7dc60c61d4ca8af568ab14cde75b36f826ab) 🙏

**Pull request merged**